src/PresentationalComponents/Inventory/Inventory.js

```diff
@@ -374,16 +374,15 @@ const Inventory = ({
               },
             },
             {
-              ...(entities?.rows?.length > filters.limit
-                ? {
-                    title: intl.formatMessage(messages.selectPage, {
-                      items: filters.limit,
-                    }),
-                    onClick: () => {
-                      onSelectRows(0, true);
-                    },
-                  }
-                : {}),
+              title: intl.formatMessage(messages.selectPage, {
+                items:
+                  entities?.rows?.length > 9
+                    ? entities.perPage
+                    : entities?.rows?.length,
```

Comment:
Imagine you'd have `per_page `set to 20 and 11 items on a page -- then this condition would be true, because 11 > 9 and you'd show that selecting page would select 20 items, which is not correct, as there are 11, aren't there?

Now from looking at this code I don't know if `entities?.rows?.length` is paginated (meaning it contains all items but at most page_size) or not paginated (contains items from all pages) -- so providing solution for both cases:

If it is the former then you could just pass
`items: entities?.rows?.length`
because this number is number of items, but at most how many fit on the page

If it's the latter then it would be fixed if you took `Math.min` of those two values right? If there are 100 items and `page_size` is 50 then minimum is 50 -- correct. If there are 15 items and `per_page` is 50, then min is 15 -- correct.

Reply:
I see what youre saying, I had figured since to code is read in cascading order the end result is what the desired result is, but the way youre looking at it is much cleaner. I'll make the necessary edits